Add AtomicCounter class to threading module

haim0n opened the pull request:
- Implemented AtomicCounter class for thread-safe increment and decrement operations.
- Ensured thread safety using a Lock.

ZeroIntensity reviewed (changes requested):
I think this needs to get discussed on [DPO](https://discuss.python.org/) first. I'm not convinced that we need this.

haim0n committed:
Add AtomicCounter class to threading module

- Implemented AtomicCounter class for thread-safe increment and decrement operations.
- Ensured thread safety using a Lock.
